packages/frontend-app-api/src/tree/instantiateAppNodeTree.ts

```diff
@@ -84,13 +85,21 @@ function resolveInputs(
         }
         throw Error(`input '${inputName}' is required but was not received`);
       }
-      return resolveInputData(input.extensionData, attachedNodes[0], inputName);
+      return {
+        extensionId: attachedNodes[0].id,
```

Comment:
This COULD actually be a full tree node here too, right? I'm wondering if we might as well go the whole way and just do it that way, just trusting people to not do insane things with the info. I'm thinking, what if when we do instantiation, there's effectively going to be a type ID and an instance ID - or anything like that. Might as well send the node then.